helper-cli/src/main/kotlin/commands/dev/RewriteTestAssetsCommand.kt

```diff
@@ -0,0 +1,209 @@
+        .file(mustExist = true, canBeFile = false, canBeDir = true, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    override fun run() {
+        val candidateFiles = findCandidateFiles(ortSourcesDir)
+        println("Found ${candidateFiles.size} candidate files.\n")
+
+        val candidateFilesByType = patchAll(candidateFiles)
+
+        candidateFilesByType.forEach { (type, files) ->
+            files.forEach { file ->
+                println("[$type] ${file.absolutePath}")
+            }
+        }
+
+        println("\nSummary\n")
+
+        candidateFilesByType.forEach { (type, files) ->
+            println("$type: ${files.size}")
+        }
+    }
+}
+
+private val REPLACE_PATTERN_REPLACEMENTS = listOf(
+    "<REPLACE_JAVA>" to "33445501",
+    "<REPLACE_OS>" to "33445502",
+    "\"<REPLACE_PROCESSORS>\"" to "33445503",
+    "\"<REPLACE_MAX_MEMORY>\"" to "33445504",
+    "<REPLACE_DEFINITION_FILE_PATH>" to "33445505",
+    "<REPLACE_ABSOLUTE_DEFINITION_FILE_PATH>" to "33445506",
+    "<REPLACE_URL>" to "http://replace/url/non/processed",
+    "<REPLACE_REVISION>" to "33445507",
+    "<REPLACE_PATH>" to "replace/path",
+    "<REPLACE_URL_PROCESSED>" to "http://replace/url/processed",
+)
+
+private val TARGET_CLASSES = setOf(
+    AnalyzerResult::class,
+    AnalyzerRun::class,
+    EvaluatorRun::class,
+    OrtResult::class,
+    ScannerRun::class,
+    ScanResult::class,
+    ProjectAnalyzerResult::class
+)
+
+private val IGNORE_PATHS_FOR_TARGET_CLASSES = mapOf(
+    AnalyzerRun::class to listOf(
+        "config",
+        "environment"
+    ),
+    EvaluatorRun::class to listOf(
+        "config",
+        "environment"
+    ),
+    OrtResult::class to listOf(
+        "analyzer/config",
+        "analyzer/environment",
+        "scanner/config",
+        "scanner/environment",
+        "evaluator/config",
+        "evaluator/environment",
+        "advisor/config",
+        "advisor/environment"
+    ),
+    ScannerRun::class to listOf(
+        "config",
+        "environment"
+    )
+)
+
+private fun findCandidateFiles(dir: File): List<File> {
+    val extensions = setOf("json", "yml", "yaml")
+
+    return dir.walkBottomUp().filter { it.isFile && it.extension in extensions }.toList().filter {
+        it.absolutePath.contains("/src/funTest/assets/") || it.absolutePath.contains("/src/test/assets/")
+    }
+}
+
+private fun patchAll(candidateFiles: Collection<File>): Map<String, List<File>> =
+    candidateFiles.groupByTo(sortedMapOf()) { it.patch() }
+
+private fun File.patch(): String = TARGET_CLASSES.firstOrNull { patch(it) }?.simpleName ?: "Unknown"
+
+@Suppress("SwallowedException")
+private fun File.patch(kClass: KClass<out Any>): Boolean =
+    try {
+        val content = readText().removeReplacePatterns()
+        val mapper = FileFormat.forFile(this).mapper
+        val value = mapper.readValue(content, kClass.java)
```

Comment:
You could probably just use `File.readValue<T>()` after making this function take a generic `T` (inline / reified would also be required).

Reply:
That's not possible, because the content of the file first needs to be patched (replace patterns) before passing it to the object mapper.